app/javascript/packages/verify-flow/steps/personal-key/download-button.tsx

```diff
@@ -0,0 +1,41 @@
+import type { MouseEventHandler } from 'react';
+import { Button } from '@18f/identity-components';
+import type { ButtonProps } from '@18f/identity-components';
+
+interface DownloadButtonProps {
+  /**
+   * Content of the downloaded file.
+   */
+  content: string;
+
+  /**
+   * File name to use for downloaded file.
+   */
+  fileName: string;
+}
+
+function DownloadButton({ content, fileName, ...buttonProps }: DownloadButtonProps & ButtonProps) {
+  const download: MouseEventHandler = (event) => {
+    buttonProps.onClick?.(event);
+
+    if ('msSaveBlob' in window.navigator) {
+      event.preventDefault();
+
+      const filename = (event.target as HTMLButtonElement).getAttribute('download');
+      const blob = new Blob([content], { type: 'text/plain' });
+
+      (window.navigator as any).msSaveBlob(blob, filename);
+    }
+  };
+
+  return (
+    <Button
+      {...buttonProps}
+      href={`data:text/plain;base64,${btoa(content)}`}
```

Comment:
we don't need the base64 here right? the server impl just URI encoded? should we make them the same for consistency? (ex change this or change the server?)

Reply:
Can you clarify what you mean by "the server impl just URI encoded" ? We initialize the page with a JSON-encoded data attribute, which wouldn't be URI-encoded.

```html
<div id="app-root" data-initial-values="{&quot;personalKey&quot;:&quot;foo bar baz&quot;}"></div>
```

However, to your point about consistent implementation, we _could_ URI-encode it with `encodeURIComponent`.

```suggestion
      href={`data:,${encodeURIComponent(content)}`}
```

Reply:
Oh, I might have misunderstood. Did you mean "server just" as is "server simply" vs. "server had already" ? In that case, I think we're on the same page as far as the alternative presented?

Reply:
Sorry, yes, exactly that I was just suggesting dropping the base64 encoding in favor of only URI encoding like we have here: https://github.com/18F/identity-idp/pull/6161/files#diff-e7b9a44412a86b2c52f074b09c7b5721c7abda57d3ce4e029a233a8e4f40ad94R14

Or updating the server to match Base64 encoding

Reply:
> Sorry, yes, exactly that

I'll update to URI encoding 👍

Reply:
Updated in 3f4d55c88.